updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -64,6 +72,23 @@ def save_job_details
                                        ))
     end
 
+    def dependency_files_for_multi_directories
```

Comment:
It looks like the result of this method needs to be memoized since it gets called twice.

In `dependency_files` this happened because we used the same `file_fetcher` twice and the `files` call was memoized. Now since we're making a file_fetcher on demand we should memoize the result. It looks like in the smoke tests it gets all the files twice, memoizing should fix that.

Reply:
Another option is to save the result from the call and parameterize `base64_dependency_files`, up to you how to fix it!

Reply:
Good catch! I will add the memoized to the `all_dependency_files` result